Tools/ardupilotwaf/chibios.py

```diff
@@ -211,14 +232,33 @@ def chibios_firmware(self):
         abin_task = self.create_task('build_abin', src=link_output, tgt=abin_target)
         abin_task.set_run_after(generate_apj_task)
 
-    bootloader_bin = self.bld.srcnode.make_node("Tools/bootloaders/%s_bl.bin" % self.env.BOARD)
+    if self.env.BOOTLOADER:
+        # copy elf
+        copy_elf_target = self.bld.srcnode.make_node("Tools/bootloaders/%s_bl.elf" % self.env.BOARD)
+        copy_elf_task = self.create_task('copy_and_rename', src=link_output, tgt=copy_elf_target)
+        copy_elf_task.set_run_after(generate_bin_task)
+        # copy bin
+        copy_bin_target = self.bld.srcnode.make_node("Tools/bootloaders/%s_bl.bin" % self.env.BOARD)
+        copy_bin_task = self.create_task('copy_and_rename', src=bin_target, tgt=copy_bin_target)
+        copy_bin_task.set_run_after(generate_bin_task)
+
     if self.bld.env.HAVE_INTEL_HEX:
-        if os.path.exists(bootloader_bin.abspath()):
+        if self.env.BOOTLOADER:
             hex_target = self.bld.bldnode.find_or_declare('bin/' + link_output.change_ext('.hex').name)
-            hex_task = self.create_task('build_intel_hex', src=[bin_target, bootloader_bin], tgt=hex_target)
-            hex_task.set_run_after(generate_bin_task)
+            hex_task = self.create_task('build_bootloader_intel_hex', src=bin_target, tgt=hex_target)
+            hex_task.set_run_after(generate_apj_task)
```

Comment:
why after the apj task and not bin task ?

Reply:
You're right, shouldn't have been changed. Fixed in ec65aa92.

Reply:
@khancyr: Any other changes you'd like me to make?